rfcXXX-github-issues-workflow.md

```diff
@@ -0,0 +1,90 @@
+---
+RFC:
+Author: Serdar Sutay <serdar@getchef.com>
+Status: Draft
+Type: Process
+---
+
+# Chef GitHub Issues Workflow
+
+This RFC codifies the workflow Chef uses to track bug reports and contributions using [GitHub Issues](https://github.com/opscode/chef/issues). Goals of this workflow are:
+
+## Terms
+
+In the context of this document below terms are used:
+
+* **Chef Users:** End users of Chef who participate in this workflow by filing **issues**.
+* **Contributors:** Users of Chef who would like to improve Chef. They participate in this workflow by submitting **contributions** to Chef.
+* **Chef Maintainers:** Developers of Chef who are responsible for one or more **subcomponents**.
+* **Lieutenants:** (Lt.) Developers of Chef who lead the maintenance of a subcomponent.
+* **Issue**: A filed **Github Issue** which can be a bug report, feature request, style change request.
+* **Contribution**: A filed **Github Pull Request** (PR).
+* **Subcomponents**: A sub-area of Chef project which is lead by a **Lt.** and a set of **maintainers**.
+
+## Goals
+
+Chef Users can:
+
+* Be aware of the cause & the next steps for the issues they have filed.
+
+Contributors can:
+
+* Identify the state of their contribution.
+
+Chef Maintainers can:
+
+* See the list of contributions they need to review in a single page.
+* See the list of issues they need to investigate in a single page.
+
+
+## Workflow
+
+![Chef GitHub Issues Workflow](./GithubIssuesWorkflow.png)
+
+
+## Notes
+
+### Common Steps for Issues & PRs
+
+The first step of Chef Github Issues Workflow is to identify the subcomponent for which the issue or PR belongs to. Once the subcomponent is identified, the rest of the workflow is the responsibility of the maintainers of the identified subcomponent.
+
+### Issues Workflow
+
+The issues workflow can be broken down into these high level steps:
+
+1. Understand the issue and determine the resolution.
+1. Identify the major version the issue should be fixed in.
+1. Label the issue to indicate completeness of **Triage** process.
+
+After this point any fix for an issue should follow the contribution workflow.
+
+#### Rules
+
+* If an issue is not commented on for more than a year, the issue is retired by closing.
+
+### Contribution Workflow
+
+The contribution workflow can be broken down into these high level steps:
+
+1. Maintainers check the contribution and give **:+1:** if the code is looking good, or leave a comment that clearly identifies what is needed for the contribution to move forward.
+1. If the simple majority of maintainers **:+1:** a PR, it is marked with `Ready to Merge` label.
+1. PRs marked with `Ready to Merge` will be merged by maintainers of the subcomponent after including the required documentation updates.
```

Comment:
Naw, I meant the past tense in there. That's what we've done in the past. I think Maintainers are the merge monkey's going forward, because they're the people with commit. For instance, maintainers Alice and Betty could both +1 a PR, and mark it Ready For Merge, and then maintainer Chuck could merge it later. I don't want to dissuade maintainers from reviewing PRs by having to take on any immediate merge workload with it.

You should write your own docs, but I'm saying that we've specifically seen with the CHANGELOG.md (and similar instances like people bumping a version in a cookbook) that you end up with constant merge conflicts if every PR changes the file. They're usually easy enough to resolve if we stay on top of the backlog and resetting the file, but the green button disappears, and then you have to wonder, "is there a real merge conflict here, or just the CHANGELOG.md"? I'm pretty sure GitHub won't trigger CI if there's a merge conflict too, which is a huge problem.

I use the merge button sometimes too, and it's convenient and all that. But I think without JIRA, we've got to step up our game on making sure our commit messages are meaningful, and that's even harder when someone does a merge pass and adds 20 merge commits on top of 20 contributions. This is really more about merging other people's branches than our own.

each of these benefits of a rebase clean up the git log and make it more usable:
* squashing/fixuping all those "i forgot a quote" commits
* not having the merge commits
* brings multiple good commits from a topic branch together in the git log, which could be spread out on long lived branches.